src/page-modules/assistant/layout.tsx

```diff
@@ -117,8 +122,14 @@ function AssistantLayout({ children, tripQuery }: AssistantLayoutProps) {
   const { urls, orgId } = getOrgData();
   const { isDarkMode } = useTheme();
 
-  if (tripQuery.transportModeFilter === null)
-    onTransportFilterChanged(setTransportModeFilters(transportModeFilter));
+  useEffect(() => {
+    if (tripQuery.transportModeFilter === null)
+      onTransportFilterChanged(setTransportModeFilters(transportModeFilter));
+  }, [
+    onTransportFilterChanged,
+    transportModeFilter,
+    tripQuery.transportModeFilter,
```

Comment:
We are sure that `tripQuery.transportModeFilter` is stable as to not change in reference every time? Won't have actual implications here other than doing unnecessary work.

Reply:
Not sure. Can have just the transportModeFilter as a dependency and ignore the eslint warning?

Reply:
could technically have `tripQuery.transportModeFilter == null` as a dependency 🤔 Or stored as an variable before

Reply:
Never used assertions in a dependency array 😲 I can try that.

Reply:
Fine to have `onTransportFilterChanged` in the dep array here?

Reply:
Need to check if the effect triggers needlessly

Reply:
Triggers twice on page load at the moment.

Reply:
@jonasbrunvoll, could you check that the effect triggers when it needs to and that it doesn't trigger needlessly, according to when the transport filter should be updated?

Reply:
@adriansberg, I checked in networks. To me it looks like useEffect is triggered once and only when there is a change in filter. I also see that queries fetching the icon SVGs are run every time the button "More choices / Less choices" is pressed, but this happens both in `adriansberg/fix-redirect-error` and in `main`. That could perhaps be improved, but that is probably a separate task of its own.